### PR TITLE
Migrate to cucumber junit 5

### PR DESCRIPTION
**Pull request by Tcharl**
Hi,

Really sorry @pascalgrimaud , I tried something: you know, I have my phases, and I'm currently in the 'traceability, unicity of the information' one (sometimes at the extreme).

This one is regarding a cucumber update to switch to a [junit 5-compatible version](https://github.com/jhipster/generator-jhipster/issues/17187), so it looks like the failed tests are expected.
I don't know how it works when I have to update the bom and the code as well: do we have to merge this one to get the snapshot version to ensure that the [cucumber templates](https://github.com/jhipster/generator-jhipster/pull/17188) works?

Maybe we could publish PR Bom's snapshots artifacts by default with a  qualifier (e.g. the PR number) in order to test it without merging in the main branch?

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
The commit msg is not nice, as we can't know what you're doing without looking the diff :-)

**Comment by DanielFran** [comment]
Updated title

**Comment by pascalgrimaud** [comment]
There is already a snapshot artifact with GitHub for main branch, if I'm correct, but I don't use it.

About the CI, you could do something similar to https://github.com/jhipster/generator-jhipster/pull/17068/files but don't forget to revert before the merge

That's why, in my opinion, splitting code into lib is a bad idea for contributors. As a core team member, it was not simple for you, I can't imagine for a new contributor.
Anyway, it's another subject.

**Comment by Tcharl** [comment]
May be just a matter of explaining more things in the CONTRIBUTING.md.
Honestly, I've spent hours to figure out how configure IJ to test generator-jhipster

File: jhipster-dependencies/pom.xml
Comment (DanielFran) [inline jhipster-dependencies/pom.xml:626]:
```suggestion
            <dependency>
                <groupId>org.testng</groupId>
                <artifactId>testng</artifactId>
                <version>${testng.version}</version>
                <scope>test</scope>
            </dependency>
            <!-- Spring Cloud Netflix dependencies are no longer managed by Spring Cloud's BOM -->
            <!-- https://github.com/spring-cloud/spring-cloud-release/wiki/Spring-Cloud-2020.0-Release-Notes#breaking-changes -->
```

**Review by DanielFran — CHANGES_REQUESTED** [review]

**Pull request merged**
